client/ayon_applications/ui/process_monitor.py

```diff
@@ -0,0 +1,697 @@
+        self._manager = manager
+        self._cleanup_type = cleanup_type  # "inactive" or "single"
+        self._process_hash = process_hash
+        self._log = getLogger(self.signature)
+
+    @Slot()
+    def run(self) -> None:
+        """Perform cleanup in background thread."""
+        self._log.debug(
+            "Starting cleanup of type: %s", self._cleanup_type)
+        try:
+            if self._cleanup_type == "inactive":
+                self._cleanup_inactive()
+            elif self._cleanup_type == "single":
+                self._cleanup_single()
+        except Exception as e:
+            self.signals.error.emit(str(e))
+
+    def _cleanup_inactive(self) -> None:
+        """Clean up inactive processes."""
+        # Get all processes to check which files to delete
+        all_processes = self._manager.get_all_process_info()
+        files_to_delete: list[Path] = []
+
+        files_to_delete.extend(
+            process.output
+            for process in all_processes
+            if (
+                not process.active
+                and (process.output and Path(process.output).exists())
+            )
+        )
+        # Delete from database
+        deleted_count = self._manager.delete_inactive_processes()
+
+        # Delete output files
+        files_deleted = 0
+        for file_path in files_to_delete:
+            # File might not exist anymore, so we use contextlib.suppress
+            with contextlib.suppress(OSError):
+                os.remove(file_path)
+                files_deleted += 1
+        self.signals.finished.emit(deleted_count, files_deleted)
+
+    def _cleanup_single(self) -> None:
+        """Clean up a single process."""
+        if not self._process_hash:
+            self.signals.error.emit("No process hash provided")
+            return
+
+        # Find the process first
+        all_processes = self._manager.get_all_process_info()
+        target_process = next(
+            (
+                process
+                for process in all_processes
+                if self._manager.get_process_info_hash(process)
+                == self._process_hash
+            ),
+            None,
+        )
+        if not target_process:
+            self.signals.error.emit("Process not found")
+            return
+
+        # Delete the output file if it exists
+        file_deleted = 0
+        if target_process.output and Path(target_process.output).exists():
+            # File might not exist anymore, so we use contextlib.suppress
+            with contextlib.suppress(OSError):
+                os.remove(target_process.output)
+                file_deleted = 1
+        # Delete from database
+        deleted = self._manager.delete_process_info(self._process_hash)
+        process_deleted = 1 if deleted else 0
+
+        self.signals.finished.emit(process_deleted, file_deleted)
+
+
+class ProcessTableModel(QtCore.QAbstractTableModel):
```

Comment:
Oh boy, you're making you life harder. You can use `QStandardItemModel` that implements most of what you do here and would probably make it faster.

Reply:
I am not really sure about that. Maybe just `headerData()` but rest of it is custom anyway. More importantly - it works and there is nothing inherently wrong about that, or is it?